ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -111,6 +112,10 @@ type RayJobSpec struct {
 	EntrypointResources string `json:"entrypointResources,omitempty"`
 	// Configurations of submitter k8s job.
 	SubmitterConfig *SubmitterConfig `json:"submitterConfig,omitempty"`
+	// Specifies the number of retries before marking this job failed.
+	// Each retry creates a new RayCluster.
+	// +kubebuilder:default:=0
+	BackoffLimit *int32 `json:"backoffLimit,omitempty"`
```

Comment:
Should we enforce `shutdownAfterJobFinishes=true` when setting backoffLimit > 0? Or should we still respect it but only for the last retry?

Reply:
Maybe the later one for now?

Reply:
sounds good, I think this is already the case then